tests/test_schemas.py

```diff
@@ -1,294 +1,645 @@
+    def test_clean_template(self, helpers, data_model):
+        # TODO: This will need to change with contexts bc they are hard coded here.
+        # Get Graph
+        graph_data_model = generate_graph_data_model(helpers, data_model_name=data_model)
+
+        # Instantiate DataModelJsonLD
+        data_model_jsonld = DataModelJsonLD(Graph=graph_data_model)
+
+        # Get empty template
+        template = data_model_jsonld.class_template()
+        assert 'sms:requiresDependency' in template
+
+        # Fill out some mock entries in the template:
+        template['@id'] == 'bts:CheckURL'
+        template['rdfs:label'] == 'CheckURL'
+        data_model_relationships=data_model_jsonld.dmr.relationships_dictionary
+
+        # Clean template
+        data_model_jsonld.clean_template(template=template, data_model_relationships=data_model_relationships)
+        
+        # Look for expected changes after cleaning
+        
+        # Check that expected JSONLD default is added
+        assert template['sms:required'] == 'sms:false'
+        assert template['sms:validationRules'] == []
+        
+        # Check that non-required JSONLD keys are removed.
+        assert 'sms:requiresDependency' not in template
+
+    @pytest.mark.parametrize("data_model", list(DATA_MODEL_DICT.keys()), ids=list(DATA_MODEL_DICT.values()))
+    @pytest.mark.parametrize("context_value", ['@id', 'sms:required'], ids=['remove_at', 'remove_sms'])
+    def test_strip_context(self, helpers, data_model, context_value):
+        # Get Graph
+        graph_data_model = generate_graph_data_model(helpers, data_model_name=data_model)
+
+        # Instantiate DataModelJsonLD
+        data_model_jsonld = DataModelJsonLD(Graph=graph_data_model)
+
+        stripped_contex = data_model_jsonld.strip_context(context_value=context_value)
+        if '@id' == context_value:
+            assert stripped_contex == ('', 'id')
+        elif 'sms:required' == context_value:
+            assert stripped_contex == ('sms', 'required')
+
+    @pytest.mark.parametrize("data_model", list(DATA_MODEL_DICT.keys()), ids=list(DATA_MODEL_DICT.values()))
+    @pytest.mark.parametrize("valid_values", [[], ['Other', 'Female', 'Male'], ['A', 'Bad', 'Entry']], ids=['Empty List', 'Disordered List', 'Incorrect List'])
+    def test_reorder_template_entries(self, helpers, data_model, valid_values):
+        # Note the way test_reorder_template_entries works, is that as long as an entry has recordings in the template
+        # even if they are incorrect, they will be corrected within this function.
+        # Get Graph
+        graph_data_model = generate_graph_data_model(helpers, data_model_name=data_model)
+
+        # Instantiate DataModelJsonLD
+        data_model_jsonld = DataModelJsonLD(Graph=graph_data_model)
+
+        # Get empty template
+        template = data_model_jsonld.class_template()
+
+        # Fill out template with some 'Sex' information
+        template['@id'] = 'Sex'
```

Comment:
I think this is a good test too. But at the same time, I also wonder what if the 'id' that you enter here is not in the schema, then what would be the expected response. Should we consider adding that in the test too?